endpoints/openrtb2/auction.go

```diff
@@ -1426,29 +1443,31 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 	return resolvedRequest, impExtInfoMap, nil
 }
 
-// parseImpInfo parses the request JSON and returns several things about the Imps
-//
-// 1. A list of the JSON for every Imp.
-// 2. A list of all IDs which appear at `imp[i].ext.prebid.storedrequest.id`.
-// 3. A list intended to parallel "ids". Each element tells which index of "imp[index]" the corresponding element of "ids" should modify.
-// 4. Any errors which occur due to bad requests. These should warrant an HTTP 4xx response.
-func parseImpInfo(requestJson []byte) (imps []json.RawMessage, ids []string, impIdIndices []int, errs []error) {
+// parseImpInfo parses the request JSON and returns impression and unmarshalled imp.ext.prebid
+func parseImpInfo(requestJson []byte) (impData []ImpData, errs []error) {
+
+	//impData := make([]ImpData, 0)
 	if impArray, dataType, _, err := jsonparser.Get(requestJson, "imp"); err == nil && dataType == jsonparser.Array {
```

Comment:
Nitpick: Do you think it'd be more useful and make for a simpler code if we use `json.Unmarshal` for the entire thing instead?
```
1446   // parseImpInfo parses the request JSON and returns impression and unmarshalled imp.ext.prebid
1447   func parseImpInfo(requestJson []byte) (impData []ImpData, errs []error) {
1448  
     +     // EXTRACT IMP EXTS     
     +     type ExtOnlyImp struct {
     +         Ext openrtb_ext.ExtImpPrebid `json:"ext"`
     +     }
     +     type requestWithExtOnlyImps struct {
     +         Imp []ExtOnlyImp `json:"imp"`
     +     }
     +     var reqOfImpsWithExts requestWithExtOnlyImps
     +     if err := json.Unmarshal(requestJson, &reqOfImpsWithExts); err != nil {
     +         return nil, append(errs, err)
     +     }
     +
     +     // EXTRACT JSON RAW IMPS
     +     type requestWithRawJsonImps struct {
     +         Imp []json.RawMessage `json:"imp"`
     +     }
     +     var reqOfRawImps requestWithRawJsonImps
     +     if err := json.Unmarshal(requestJson, &reqOfRawImps); err != nil {
     +         return nil, append(errs, err)
     +     }
     +
     +     // PUT THEM TOGETHER
     +     for i:=0;i< len(reqOfImpsWithExts); i++ {
     +         newImpData := ImpData{reqOfRawImps[i], reqOfImpsWithExts[i]}
     +         impData = append(impData, newImpData)
     +     }
1449 -     //impData := make([]ImpData, 0)
1450 -     if impArray, dataType, _, err := jsonparser.Get(requestJson, "imp"); err == nil && dataType == jsonparser.Array {
1451 -         _, err = jsonparser.ArrayEach(impArray, func(imp []byte, _ jsonparser.ValueType, _ int, err error) {
1452 -             impExtData, _, _, err := jsonparser.Get(imp, "ext", "prebid")
1453 -             var impExtPrebid openrtb_ext.ExtImpPrebid
1454 -             if impExtData != nil {
1455 -                 if err := json.Unmarshal(impExtData, &impExtPrebid); err != nil {
1456 -                     errs = append(errs, err)
1457 -                 }
1458 -             }
1459 -             newImpData := ImpData{imp, impExtPrebid}
1460 -             impData = append(impData, newImpData)
1461 -         })
1462 -     }
1463       return
1464   }
1465
endpoints/openrtb2/auction.go
```

Reply:
This is a good idea from the first glance. I started to implement it and realize this is similar to full request unmarshalling we do later in code. 
Idea here is to find very specific part of imp.ext, which I think will be present in minority of cases. 
I honestly don't have a strong opinion about this. Need to discuss it with a team.